framework/src/problems/FEProblemBase.C

```diff
@@ -3954,6 +3961,68 @@ FEProblemBase::execute(const ExecFlagType & exec_type)
 
   // Return the current flag to None
   setCurrentExecuteOnFlag(EXEC_NONE);
+
+  if (_uo_aux_state_check && !_checking_uo_aux_state)
+  {
+    // we will only check aux variables and postprocessors
+    // checking more reporter data can be added in the future if needed
+    std::unique_ptr<NumericVector<Number>> x = _aux->currentSolution()->clone();
+    DenseVector<Real> pp_values = getReporterData().getAllRealReporterValues();
+
+    // call THIS execute one more time for checking the possible states
+    _checking_uo_aux_state = true;
+    FEProblemBase::execute(exec_type);
+    _checking_uo_aux_state = false;
+
+    const Real check_tol = 1e-8;
+
+    const Real xnorm = x->l2_norm();
+    *x -= *_aux->currentSolution();
+    if (x->l2_norm() > check_tol * xnorm)
+    {
+      const auto & sys = _aux->system();
+      unsigned int n_vars = sys.n_vars();
+      std::multimap<Real, std::string, std::greater<Real>> ordered_map;
+      for (unsigned int i = 0; i < n_vars; i++)
```

Comment:
```suggestion
      for (const auto i : make_range(n_vars))
```

Reply:
I think `n_vars` is in unsigned int, so the old way is fine. But I will make this change.

Reply:
The change is robust against any changes to the type of `n_vars` and it ensures no one does anything funny with the loop counter